torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -538,12 +538,28 @@ RegisterOperators reg({
     Operator(
         prim::fork,
         [](Node* node) {
+          int n_inputs = node->inputs().size();
           Code code(node->g(attr::Subgraph));
           JIT_ASSERT(node->blocks().size() == 0);
           JIT_ASSERT(node->hasAttribute(attr::Subgraph));
           return [=](Stack& stack) {
-            InterpreterState(code).run(stack);
-            push(stack, Future(pop(stack)));
+            // Move inputs to a separate stack
+            auto copied_stack = std::make_shared<Stack>();
+            std::vector<IValue> inputs(n_inputs);
+            for (int i = 0; i < n_inputs; ++i) {
+              pop(stack, inputs[i]);
+            }
+            for (int i = n_inputs - 1; i >= 0; --i) {
+              push(*copied_stack, inputs[i]);
```

Comment:
`Stack` is just `std::vector<IValue>`, so why do we need to intermediate step? You could simply push it directly to `(*copied_stack)[i]`.